gqcp/tests/Basis/SpinorBasis/SpinResolvedOrbitalSpace_test.cpp

```diff
@@ -0,0 +1,119 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#define BOOST_TEST_MODULE "SpinResolvedOrbitalSpace"
+
+#include <boost/test/unit_test.hpp>
+
+#include "Basis/SpinorBasis/OrbitalSpace.hpp"
+#include "Basis/SpinorBasis/SpinResolvedOrbitalSpace.hpp"
+
+
+// Create some shortcuts to be used in the following tests.
+const auto occ = GQCP::OccupationType::k_occupied;
+const auto act = GQCP::OccupationType::k_active;
+const auto virt = GQCP::OccupationType::k_virtual;
+
+const auto alpha = GQCP::Spin::alpha;
+const auto beta = GQCP::Spin::beta;
+
+
+/**
+ *  Check whether the constructor works as expected.
+ */
+BOOST_AUTO_TEST_CASE(constructor) {
+
+    // Create potential sets of indices for all occupation types.
+    const std::vector<size_t> occupied_indices {0, 1, 2};
+    const std::vector<size_t> active_indices {3};
+    const std::vector<size_t> virtual_indices {4, 5, 6, 7};
+    const std::vector<size_t> all_indices {0, 1, 2, 3, 4, 5, 6, 7};
+
+    // Construct an alpha and beta orbital space.
+    const GQCP::OrbitalSpace orbital_space_a {occupied_indices, active_indices, virtual_indices};
+    const GQCP::OrbitalSpace orbital_space_b {occupied_indices, active_indices, virtual_indices};
+
+    //Create the spin resolved orbital space from the alpha and beta component.
+    const GQCP::SpinResolvedOrbitalSpace orbital_space {orbital_space_a, orbital_space_b};
+
+    // Test whether or not the indices match. Some are tested for alpha, some for beta.
+    BOOST_CHECK(orbital_space.alpha().indices() == all_indices);
+    BOOST_CHECK(orbital_space.beta().indices(occ) == occupied_indices);
+    BOOST_CHECK(orbital_space.alpha().indices(act) == active_indices);
+    BOOST_CHECK(orbital_space.beta().indices(virt) == virtual_indices);
+
+    BOOST_CHECK(orbital_space.alpha().numberOfOrbitals() == 8);
+    BOOST_CHECK(orbital_space.beta().numberOfOrbitals(occ) == 3);
+    BOOST_CHECK(orbital_space.alpha().numberOfOrbitals(act) == 1);
+    BOOST_CHECK(orbital_space.beta().numberOfOrbitals(virt) == 4);
+}
```

Comment:
This functionality is already tested in `OrbitalSpace`, and since `SpinResolvedOrbitalSpace` is just a composition of two of them, you can leave this in, or you can remove it. It's up to you.

Reply:
I'll remove it if it's unnecessary. Since we discourage code duplication as much as possible in the code itself, it makes sense to do so as well in the tests! :)